turbopack/crates/turbo-tasks-macros/src/function_macro.rs

```diff
@@ -102,3 +102,12 @@ pub fn function(args: TokenStream, input: TokenStream) -> TokenStream {
     }
     .into()
 }
+
+// `&self` is `self: Vc<Self>` with self.await? so it's mutable
```

Comment:
nit:
```suggestion
/// Computes whether the task is statically immutable based on the signature.
/// - if the task is `async` we assume it is reading some other task.
/// - if a task accepts `&self` then we know it read `Vc<Self>` in the generated calling code.
///  See also: turbopack/crates/turbo-tasks/src/task/function.rs for the binding code.
```

This also makes me wonder if we should include the 'ignores_self' analysis here?